scripts/cleanup_additional_repos.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+#
+# Clean up additional YUM repositories, typically used for security patches.
+# The format of ADDITIONAL_YUM_REPOS is: "repo=patches-repo,name=Install patches,baseurl=http://amazonlinux.$awsregion.$awsdomain/xxxx,priority=10"
+# Multiple yum repos can be specified, separated by ';'
+
+if [ -z "${ADDITIONAL_YUM_REPOS}" ]; then
+  echo "no additional yum repo, skipping"
+  exit 0
+fi
+
+
+AWK_CMD='
+BEGIN {RS=";";FS=","}
+{
+  delete vars; 
+  for(i = 1; i <= NF; ++i) { 
+    n = index($i, "="); 
+    if(n) { 
+      vars[substr($i, 1, n-1)] = substr($i, n + 1)
+    }
+  } 
+  Repo = "/etc/yum.repos.d/"vars["repo"]".repo"
+}
+{cmd="rm -f " Repo; system(cmd)}
+'
+sudo awk "$AWK_CMD" <<< "${ADDITIONAL_YUM_REPOS}"
```

Comment:
This is really hard to read and maintain. Does Packer support high level programming language, i.e. Python?

Reply:
we need to install python runtime first.
TBH, it's similar to python programing, you'll have to write similar logic in python as well for parse such logic into a yum repo configuration.

Reply:
To be more specific. 

Reader/maintainers needs to understand and remember awk `substr` is one based not the standard zero based.  https://www.gnu.org/software/gawk/manual/html_node/String-Functions.html#FOOT48
```
n = index($i, "="); 
    if(n) { 
      vars[substr($i, 1, n-1)] = substr($i, n + 1)
    }
```
Comparing with the Python version.
```
pair = split(field, '='):
vars[pair[0]] = dict[pair[1]]
```

Reply:
to be fair..that's how shell sucks right....like 0 means true, 1 means false 🤣